portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java

```diff
@@ -15,30 +16,44 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.time.format.DateTimeFormatter;
+import java.time.*;  
 
-/** Servlet responsible for listing tasks. */
+
+/** Servlet responsible for listing comments. */
 @WebServlet("/comments-list")
 public class CommentsListServlet extends HttpServlet {
-
+  
+  private DatastoreService datastore;
+  
+  @Override
+  public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  } 
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments = Integer.parseInt(request.getParameter("max")); 
     Query query = new Query("Comment").addSort("timestamp_millis", SortDirection.DESCENDING);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    FetchOptions fetchOps = FetchOptions.Builder.withDefaults(); 
+    
+    if(maxComments != -1){
+      fetchOps = FetchOptions.Builder.withLimit(maxComments); 
+    }
+    
+    for (Entity entity : results.asIterable(fetchOps)) {
       long id = entity.getKey().getId();
       String content = (String) entity.getProperty("content");
       long timestamp_millis = (long) entity.getProperty("timestamp_millis");
-
-      Comment comment = new Comment(id, content, timestamp_millis);
+      String currentDate = (String) entity.getProperty("currentDate");
```

Comment:
I think @jalexanderqed is recommending the space between the cast type `(String)` and the expression `entity(...)` be removed (here and throughout).

Reply:
This isn't fixed.